pkg/apis/stable/v1alpha1/fleetautoscaler.go

```diff
@@ -105,30 +105,10 @@ type FleetAutoscalerStatus struct {
 	// lastScaleTime is the last time the FleetAutoscaler scaled the attached fleet,
 	// +optional
 	LastScaleTime *metav1.Time `json:"lastScaleTime"`
-
-	// AbleToScale indicates that we can access the target fleet
-	AbleToScale bool `json:"ableToScale"`
```

Comment:
Why did you remove these flags?
I thought they should be pretty helpful for monitoring autoscaling state

Reply:
This is why I wanted you to look :smile: I made recorded events for these items, so that we could see how often they happened, have a timestamp for them, make it easy to see it in the `kubectl describe history` - and these events can also be used to power stats as well (they are globally collectable).

An example probably will help, so here is a FAS, which initially started with a Fleet that couldn't be found, and then was hitting the max limits for the autoscalers. You can see how often FailedGetFleet happened, and over what period, as well as when the last one was.

```
root@7ea86f2112be:/go/src/agones.dev/agones# kubectl describe fas
Name:         simple-udp-autoscaler
...
Spec:
  Fleet Name:  simple-udp
  Policy:
    Buffer:
      Buffer Size:   2
      Max Replicas:  6
      Min Replicas:  0
    Type:            Buffer
Status:
  Current Replicas:  5
  Desired Replicas:  6
  Last Scale Time:   2018-11-16T20:24:57Z
Events:
  Type     Reason            Age              From                        Message
  ----     ------            ----             ----                        -------
  Warning  FailedGetFleet    2m (x2 over 3m)  fleetautoscaler-controller  could not fetch fleet: simple-udp
  Normal   AutoScalingFleet  1m               fleetautoscaler-controller  Scaling fleet simple-udp from 2 to 3
  Normal   AutoScalingFleet  1m               fleetautoscaler-controller  Scaling fleet simple-udp from 3 to 5
  Normal   AutoScalingFleet  25s              fleetautoscaler-controller  Scaling fleet simple-udp from 5 to 6
  Warning  ScalingLimited    25s              fleetautoscaler-controller  Scaling fleet simple-udp was limited to maximum size of 6
```
Once i did this, the `AbleToScale` and `ScalingLimited` seemed redundant to me, so I removed them, as I didn't think they were adding any value at that point.

But like I said, I took some liberties :smile: , mainly as a sacrificial draft, to get the feedback to see if it made sense in the grand scheme, and be able to show an example of what I was thinking.

Does that make sense? Would love to hear your thoughts.

Reply:
I added those two flags following the example of HPA.

I personally believe that it makes sense for those to be part of status because... They are describing the current status.

The event log has a different purpose imo, to show the history.

I can also argue that If we follow this events logic than why have any status members at all? You can retrieve all of them from events anyway 😋

Reply:
That's a totally fair point, although I don't see them here:
https://github.com/kubernetes/api/blob/master/autoscaling/v1/types.go#L53

There really isn't any downside in having both. I'll put them back in then! :+1:

Reply:
Just digging deeper to see if maybe there is a better pattern - I see the above values (`AbleToScale`, `ScalingLimited`) in `HorizontalPodAutoscalerCondition` - but can't find if it's used anywhere. 

Pods do have a `[]PodCondition` in `PodStatus` - maybe the plan was to do something like that, but it never eventuated. This is probably overkill for our FAS though. WDYT?

Reply:
Look at HorizontalPodAutoscalerConditionType 😋

It's a slightly different implemention pattern. My go was not good enough, so I simplified things a bit 😁

Reply:
Yep - looks like we are in furious agreement 👍😀😁

I'll put it back and in and add the tests back. Be good to get this in for 0.6.0.rc next week.

Reply:
Furious agreement 🤣🤣🤣

Reply:
Done!